xen-hvm-stubdom-legacy.spec.in

```diff
@@ -0,0 +1,291 @@
+
+Patch1020: patch-stubdom-linux-config-qubes-gui.patch
+Patch1021: patch-stubdom-linux-libxl-do-not-force-qdisk-backend-for-cdrom.patch
+Patch1022: patch-xen-acpi-slic-support.patch
+
+Patch1100: patch-fix-errors-on-ambiguous-python-shebang.patch
+
+BuildRequires: xen-devel = %{epoch}:%{version}-%{release}
+BuildRequires: bison
+BuildRequires: flex
+BuildRequires: autoconf
+BuildRequires: automake
+BuildRequires: gettext
+BuildRequires: glibc-devel
+BuildRequires: gnutls-devel
+BuildRequires: openssl-devel
+BuildRequires: python2-devel
+BuildRequires: ncurses-devel gtk2-devel libaio-devel nettle-devel
+BuildRequires: pciutils-devel
+BuildRequires: libuuid-devel
+BuildRequires: acpica-tools
+BuildRequires: bzip2-devel xz-devel
+BuildRequires: yajl-devel
+BuildRequires: cmake
+BuildRequires: edk2-ovmf
+BuildRequires: dev86
+
+# so that x86_64 builds pick up glibc32 correctly
+BuildRequires: /usr/include/gnu/stubs-32.h
+
+%description
+Legacy stubdom files for Xen
+
+
+%prep
+%autosetup -p1 -n xen-%{upstream_version}
+
+# stubdom sources
+cp -v %{SOURCE10} %{SOURCE11} %{SOURCE12} %{SOURCE13} %{SOURCE14} stubdom
+cp -v %{SOURCE15} %{SOURCE16} %{SOURCE18} stubdom
+
+# qubes specific parts of stubdom
+tar xf %{SOURCE33}
+mkdir tools/qubes-gui/
+cp -a gui-agent-xen-hvm-stubdom/* tools/qubes-gui/
+
+tar xf %{SOURCE36}
+cp -a gui-common/include/qubes-gui*.h tools/qubes-gui/include/
+make -C tools/qubes-gui clean
+
+tar xf %{SOURCE34}
+cp -a core-vchan-xen/vchan tools/
+make -C tools/vchan -f Makefile.stubdom clean
+sed -e 's/ioemu-qemu-xen/qemu-xen-traditional/g' tools/qubes-gui/gui-agent-qemu/qemu-glue.patch | patch -p1
+
+tar xf %{SOURCE35}
+cp -a stubdom-dhcp/* tools/qemu-xen-traditional/
+patch -d tools/qemu-xen-traditional -p4 < stubdom-dhcp/lwip-dhcp-qemu-glue.patch
+
+
+%build
+
+%define efi_flags EFI_VENDOR=qubes
+mkdir -p dist/install/boot/efi/efi/qubes
+
+export XEN_VENDORVERSION="-%{release}"
+export EXTRA_CFLAGS_XEN_TOOLS="$RPM_OPT_FLAGS"
+export EXTRA_CFLAGS_QEMU_TRADITIONAL="$RPM_OPT_FLAGS"
+export EXTRA_CFLAGS_QEMU_XEN="$RPM_OPT_FLAGS"
+export PYTHON="/usr/bin/python2"
+export KCONFIG_CONFIG=%{SOURCE3}
+export XEN_CONFIG_EXPERT=y
+export PATH="/usr/bin:$PATH"
+
+cp -f %{SOURCE3} xen/.config
+
+autoreconf
+./configure \
+    --prefix=%{_prefix} \
+    --libdir=%{_libdir} \
+    --libexecdir=%{_libexecdir} \
+    --enable-vtpm-stubdom \
+    --enable-vtpmmgr-stubdom \
+    --disable-seabios \
+    --disable-xen \
+    --disable-docs \
+    --disable-largefile \
+    --disable-githttp \
+    --disable-monitors\
+    --disable-ocamltools \
+    --disable-xsmpolicy \
+    --disable-ovmf \
+    --disable-blktap2 \
+    --disable-rombios \
+    --disable-ipxe \
+    --disable-systemd \
+    --disable-9pfs
+
+export RPM_OPT_FLAGS_RED=`echo $RPM_OPT_FLAGS | sed -e 's/-m64//g' -e 's/--param=ssp-buffer-size=4//g' -e's/-fstack-protector-strong//'`
+export EXTRA_CFLAGS_XEN_TOOLS="$RPM_OPT_FLAGS_RED"
+
+# tools
+make %{?_smp_mflags} prefix=/usr tools
```

Comment:
Have you tried without this step?

Reply:
Yes it's not working due to xentoolcore_internal.h.